kubectl-fdb/cmd/analyze.go

```diff
@@ -0,0 +1,392 @@
+				if err != nil {
+					errs = append(errs, err)
+				}
+			}
+
+			if len(errs) > 0 {
+				var errMsg strings.Builder
+
+				for _, err := range errs {
+					errMsg.WriteString("\n")
+					errMsg.WriteString(err.Error())
+				}
+
+				return fmt.Errorf(errMsg.String())
+			}
+
+			return nil
+		},
+		Example: `
+# Analyze the cluster "sample-cluster-1" in the current namespace
+kubectl fdb analyze sample-cluster-1
+
+# Analyze the cluster "sample-cluster-1" in the namespace "test-namespace"
+kubectl fdb -n test-namespace analyze sample-cluster-1
+
+# Analyze the cluster "sample-cluster-1" in the current namespace and fixes issues
+kubectl fdb analyze --auto-fix sample-cluster-1
+`,
+	}
+	cmd.SetOut(o.Out)
+	cmd.SetErr(o.ErrOut)
+	cmd.SetIn(o.In)
+
+	cmd.Flags().Bool("auto-fix", false, "defines if the analyze tasks should try to fix the found issues (e.g. replace Pods).")
+	cmd.Flags().Bool("all-clusters", false, "defines all clusters in the given namespace should be analyzed.")
+	// We might want to move this into the root cmd if we need this in multiple places
+	cmd.Flags().Bool("no-color", false, "Disable color output.")
+
+	o.configFlags.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func printStatement(cmd *cobra.Command, line string, err bool) {
+	if err {
+		color.Set(color.FgRed)
+		cmd.PrintErrf("✖ %s\n", line)
+		color.Unset()
+		return
+	}
+
+	color.Set(color.FgGreen)
+	cmd.Printf("✔ %s\n", line)
+	color.Unset()
+}
+
+// TODO (johscheuer): check if we have enough resources.
+func replaceFailedInstances(cluster *fdbtypes.FoundationDBCluster, instanceIDs []string) {
+	for _, instanceID := range instanceIDs {
+		markedForRemoval := false
+		for _, removal := range cluster.Spec.InstancesToRemove {
+			if removal != instanceID {
+				continue
+			}
+
+			markedForRemoval = true
+		}
+
+		for _, removal := range cluster.Spec.InstancesToRemoveWithoutExclusion {
+			if removal != instanceID {
+				continue
+			}
+
+			markedForRemoval = true
+		}
+
+		if !markedForRemoval {
+			cluster.Spec.InstancesToRemove = append(cluster.Spec.InstancesToRemove, instanceID)
+		}
+	}
+}
+
+func analyzeCluster(cmd *cobra.Command, kubeClient client.Client, clusterName string, namespace string, autoFix bool, force bool) error {
+	foundIssues := false
+	var cluster fdbtypes.FoundationDBCluster
+	err := kubeClient.Get(ctx.Background(), client.ObjectKey{
+		Namespace: namespace,
+		Name:      clusterName,
+	}, &cluster)
+
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return fmt.Errorf("could not get cluster: %s/%s", namespace, clusterName)
+		}
+		return err
+	}
+
+	cmd.Printf("Checking cluster: %s/%s\n", namespace, clusterName)
+	patch := client.MergeFrom(cluster.DeepCopy())
+
+	// 1. Check if the cluster is available and healthy!
+	if cluster.Status.Health.Available {
+		printStatement(cmd, "Cluster is available", false)
+	} else {
+		foundIssues = true
+		printStatement(cmd, "Cluster is not available", true)
+	}
+
+	if cluster.Status.Health.Healthy {
```

Comment:
I'm not sure if this check is worth including. The `healthy` flag in FDB covers a lot of things that aren't obvious from the name, and this could lead to noise in the output.

Reply:
That's right, that was also one of my concerns and is a good point. I will remove it.